FWCore/TestProcessor/src/TestProcessor.cc

```diff
@@ -638,34 +630,25 @@ namespace edm {
 
         LumiTransitionInfo transitionInfo(*lumiPrincipal, es, nullptr);
 
-        std::vector<edm::SubProcess> emptyList;
-
         //To wait, the ref count has to be 1+#streams
         {
           FinalWaitingTask streamLoopWaitTask{taskGroup_};
 
           using Traits = OccurrenceTraits<LuminosityBlockPrincipal, BranchActionStreamEnd>;
 
-          endStreamsTransitionAsync<Traits>(WaitingTaskHolder(taskGroup_, &streamLoopWaitTask),
-                                            *schedule_,
-                                            preallocations_.numberOfStreams(),
-                                            transitionInfo,
-                                            serviceToken_,
-                                            emptyList,
-                                            false);
+          for (unsigned int i = 0; i < preallocations_.numberOfStreams(); ++i) {
+            schedule_->processOneStreamAsync<Traits>(
```

Comment:
I think this loop has a problem. Each iteration creates a new `WaitingTaskHolder`, while the old code made 1 which lived for the lifetime of the loop. If the thread running the loop was stalled, then the already started scheduled could finish their jobs an cause all the previously created `WaitingTaskHolders` to go away thereby starting the held task BEFORE this loop finishes.